### PR TITLE
lib/connections: Correctly add port to portless tcp:// URLs (fixes #3115)

### LINKED ISSUE #3115: Connections with static addresses without port fail
An address like `tcp://foo.example.com` gets interpreted as `tcp://:22000` which in turn becomes 127.0.0.1:22000 or just a broken address, depending on OS.

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Fixes a bug.

### Testing

Manually, it works now.

### CODE REVIEW

File: lib/connections/tcp_listen.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func fixupPort(uri *url.URL) *url.URL {
 	host, port, err := net.SplitHostPort(uri.Host)
 	if err != nil && strings.HasPrefix(err.Error(), "missing port") {
 		// addr is on the form "1.2.3.4"
-		copyURI.Host = net.JoinHostPort(host, "22000")
```
Comment (calmh) [inline lib/connections/tcp_listen.go]: When err is non-nil `host` is empty so we can't use that.

**Comment by calmh** [comment]
@st-review merge it, i'm good for it

**Comment by st-review** [comment]
:ok_hand: Merged as 8e060e23e345c8135b149e570444b515607c5205. Thanks, @calmh!

**Comment by AudriusButkevicius** [comment]
👍

**Pull request closed without merging**